server/src/test/java/io/spine/examples/shareaware/server/e2e/given/SharePurchaseTestEnv.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    public static WalletBalance walletBalanceWith(Money amount, WalletId walletId) {
+        return WalletBalance
+                .newBuilder()
+                .setId(walletId)
+                .setBalance(amount)
+                .vBuild();
+    }
+
+    public static SharesPurchased sharesPurchasedAfter(PurchaseShares command) {
+        return SharesPurchased
+                .newBuilder()
+                .setPurchaser(command.getPurchaser())
+                .setPurchaseProcess(command.getPurchaseProcess())
+                .setShare(command.getShare())
+                .setSharesAvailable(command.getQuantity())
+                .vBuild();
+    }
+
+    public static WalletBalance balanceAfterTeslaPurchase(Money sharePrice,
+                                                          WalletBalance balance) {
+        Money reducedBalance = subtract(balance.getBalance(), sharePrice);
+        return balance
+                .toBuilder()
+                .setBalance(reducedBalance)
+                .vBuild();
+    }
+
+    public static InvestmentView investmentAfterTeslaPurchase(Share tesla, UserId user) {
+        InvestmentId id = investmentId(user, tesla.getId());
+        return InvestmentView
+                .newBuilder()
+                .setId(id)
+                .setSharesAvailable(1)
+                .vBuild();
+    }
+
+    public static WithdrawMoney withdrawAllMoney(WalletBalance balance, WalletId wallet) {
+        return WithdrawMoney
+                .newBuilder()
+                .setWithdrawalProcess(WithdrawalId.generate())
+                .setWallet(wallet)
+                .setRecipient(IBAN)
+                .setAmount(balance.getBalance())
+                .vBuild();
+    }
+
+    public static MoneyWithdrawn moneyWithdrawnAfter(WithdrawMoney command,
+                                                     WalletBalance currentBalance) {
+        Money reducedBalance = subtract(currentBalance.getBalance(), command.getAmount());
+        return MoneyWithdrawn
+                .newBuilder()
+                .setWithdrawalProcess(command.getWithdrawalProcess())
+                .setWallet(command.getWallet())
+                .setCurrentBalance(reducedBalance)
+                .vBuild();
+    }
+
+    private static WithdrawalOperationId operationId(PurchaseId id) {
+        return WithdrawalOperationId
+                .newBuilder()
+                .setPurchase(id)
+                .vBuild();
+    }
+
+    private static InvestmentId investmentId(UserId user, ShareId share) {
+        return InvestmentId
+                .newBuilder()
+                .setOwner(user)
+                .setShare(share)
+                .vBuild();
+    }
+
+    public static Share chooseTeslaShareFrom(Collection<Share> shares) {
+        Optional<Share> tesla = shares
+                .stream()
+                .filter((share -> share.getCompanyName()
+                                       .equals("Tesla Inc.")))
```

Comment:
Can we make it less prone to errors and typos? Maybe, `.toLowerCase().contains("tesla")`?